### ISSUE TITLE
Support for creating Link to Text Fragments with `textStart,textEnd` format

### ISSUE THREAD

**Issue by Qhilm** [debanjum/deep-dive#4]
Instead of generating a link like this when highlighting a large portion of text:

`https://blog.chromium.org/2019/12/chrome-80-content-indexing-es-modules.html#:~:text=The%20Content%20Indexing%20API%20provides%20metadata%20about%20content%20that%20your%20web%20app%20has%20already%20cached.%20More%20specifically,%20it%20stores%20URLs%20for%20HTML%20documents%20that%20display%20stored%20media.%20The%20new%20API%20lets%20you%20add,%20list,%20and%20remove%20resources.%20Browsers%20can%20use%20the%20information%20in%20the%20index%20to%20display%20a%20list%20of%20offline-capable%20content.%20Here's%20what%20it%20looks%20like%20in%20Chrome%2080.`

Generate a much shorter link like this:

`https://blog.chromium.org/2019/12/chrome-80-content-indexing-es-modules.html#:~:text=The%20Content%20Indexing%20API,like%20in%20Chrome%2080.`

There are some risks of breakage if the text has some text snippets multiple times, of course, but I think it's worth it.

Long discussion on this topic here, if interested: https://github.com/mozilla/standards-positions/issues/194

**Comment by roke-julian-lockhart** [comment]
Remember that both are useful, since the link contains what is practically an embed of the text, which is useful when generating link previews without having to load the external site.

**Comment by Qhilm** [comment]
Absolutely. The format suggested here should only be used when a certain text size threshold is reached, to keep the link from becoming unwieldy...

**Comment by debanjum** [comment]
Yeah, having long deep links can be a little annoying for long highlights. But switching to startText, endText format for long text will be more trouble than it's worth for now, IMO. As once we have that, folks will want a settings page with option to choose deep link format or set link length to switch between the two formats. Also the higlighting logic will need a more complex update than the current simple `window.findq mechanism it uses to jump to section and highlight it

**Comment by Qhilm** [comment]
I personally think it would be very useful, but I wouldn't dare say it's "worth it" – you're the one putting in all the effort, for free =) And thank you for that btw.

**Comment by debanjum** [comment]
I'm open to a PR to add support for this. But it's unlikely I'll get time myself this year to implement it
